File: app/src/main/java/com/testapp/testapp/model/utils/DistanceFormatter.java
package com.testapp.testapp.model.utils;

import java.util.Locale;

/**
 * Created by troll on 05.11.2017.
 */

public class DistanceFormatter {

    public static String fromMetersToKilometersFormat(int distanceInMeters){
        int precise = 10;
        double distanceInKilometers = (double) distanceInMeters / Constants.KILOMETER_IN_METERS;
        distanceInKilometers = distanceInKilometers * precise;
        int roundedDistance = (int) Math.round(distanceInKilometers);
        distanceInKilometers = (double) roundedDistance / precise;
        return String.format(Locale.US,"%.1f", distanceInKilometers);
    }
}

File: app/src/main/java/com/testapp/testapp/kotlin/model/utils/DistanceFormatter.kt
package com.testapp.testapp.kotlin.model.utils

import com.testapp.testapp.Constants
import java.util.*

class DistanceFormatter {

    companion object {
        fun fromMetersToKilometersFormat(distanceInMeters: Int): String {
            val precise = 10
            var distanceInKilometers: Double = distanceInMeters.toDouble() / Constants.UI.KILOMETER_IN_METERS
            distanceInKilometers *= precise
            val roundedDistance: Int = Math.round(distanceInKilometers).toInt()
            distanceInKilometers = roundedDistance.toDouble() / precise
            return String.format(Locale.US, "%.1f", distanceInKilometers)
        }
    }
}